frontend/src/components/form/ControlledTextInput.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useRef } from "react";
+import { useController } from "react-hook-form";
+
+import { AddressInputProps } from "../../types/form.type";
+
+export const ControlledTextInput = ({
```

Comment:
This sounds like it can be used for general purposes but has to be specific to the address form, right?
Why don't we name it `AddressInput` as the name of the props of this component is `AddressInputProps`?